### PR TITLE
[4.0] Featured view: corrected rss feed func

### PR DESCRIPTION
**Pull request by bahl24**
Pull Request for Issue - Syndication Module in frontend not working .

### Testing Instructions
screenshot from 2019-02-27 03-13-06
Click on any RSS button to get an error

### Expected result
screenshot from 2019-02-27 03-14-42
XML version

### Actual result
Error

### CODE REVIEW

**Comment by infograf768** [comment]
This is only broken for the featured view with error
`Call to undefined method Joomla\Component\Content\Site\View\Featured\FeedView::escape()`
Due to the fact that `escape()` is not present in `AbstractView` while it is present in `HtmlView`.

I tested this OK and also if I replace the line by
`$title = htmlspecialchars($row->title, ENT_QUOTES, 'UTF-8');`

Note that feedview for a category (blog) is extending `HtmlView` where we do have `$title = $this->escape($item->$titleField);`

No idea if the correct solution is to mimick `CategoryFeedView` for featured view or not.

**Comment by bahl24** [comment]
@infograf768 I have made the changes you suggested and escape function is the same as `return htmlspecialchars($var, ENT_QUOTES, $this->_charset);` so it doesn't matter and seems correct to me.

File: components/com_content/View/Featured/FeedView.php [hunk 1]
```diff
@@ -50,7 +50,7 @@ public function display($tpl = null)
 		foreach ($rows as $row)
 		{
 			// Strip html from feed item title
-			$title = $this->escape($row->title);
+			$title = $title = htmlspecialchars($row->title, ENT_QUOTES, 'UTF-8');
```
Comment (alikon) [inline components/com_content/View/Featured/FeedView.php]: one `$title` is enough

**Review by alikon — CHANGES_REQUESTED** [review]

File: components/com_content/View/Featured/FeedView.php (hunk 1 above)
Comment (bahl24) [reply components/com_content/View/Featured/FeedView.php]: Yes there is no difference between the two, `$title = htmlspecialchars($row->title, ENT_QUOTES, 'UTF-8');` should stay or the other one?
I think ` htmlspecialchars` works fine.

Comment (alikon) [reply components/com_content/View/Featured/FeedView.php]: i mean
`$title = $title = htmlspecialchars($row->title, ENT_QUOTES, 'UTF-8'); `
should be
`$title = htmlspecialchars($row->title, ENT_QUOTES, 'UTF-8'); `

Comment (bahl24) [reply components/com_content/View/Featured/FeedView.php]: oh, my mistake. Thanks for pointing out.

**Comment by bahl24** [comment]
@alikon Kindly test this

**Comment by bahl24** [comment]
@infograf768 Kindly test this

**Comment by infograf768** [comment]
Restarted drone

**Comment by bahl24** [comment]
@infograf768 I think there is some issue in 4.0 branch itself as build of many PR is failing.

**Comment by bahl24** [comment]
@infograf768 @alikon Restarted as drone is back on track,kindly test

**Comment by bahl24** [comment]
@infograf768 Kindly test this, I think the changes have been finalised

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 41c00b83f128911091b34536dcaed61ea6e4e0b9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24028">issues.joomla.org/tracker/joomla-cms/24028</a>.</sub>

**Comment by bahl24** [comment]
@Quy Kindly test

**Comment by bahl24** [comment]
@Quy @alikon  Kindly test

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on 540dd2f8954c497404b9897bbffbb7e5533213b9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24028">issues.joomla.org/tracker/joomla-cms/24028</a>.</sub>

**Comment by ghost** [comment]
@infograf768 can you please retest?

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24028">issues.joomla.org/tracker/joomla-cms/24028</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**